### PR TITLE
Dedicated Timing Plots - New Settings for uGT and Eta/Phi Map - L1T & Online DQM

### PR DESCRIPTION
**Pull request by ghost**
My task in JIRA is : ([https://its.cern.ch/jira/browse/CMSLITDPG-165]).

First of all, I would like to mention that I added a new and more convenient way to implement the necessary algorithm trigger bits (L1_FirstCollisionInTrain, L1_LastCollisionInTrain) for producing pre- post- firing plots using them as parameters for the first bunch, last bunch and isolated bunch. After that, I produced a new module for producing pre- post- firing plots with a second way using eta/phi map for L1T objects (muon, jet, tau, egamma, MET) for every BX as well as for the first bunch, last bunch and isolated bunch.

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by thomreis** [comment]
Hi @dtsitson 
how is this PR related to #21744 ? Have the comments there all been addressed?

**Comment by dtsitson** [comment]
No, I did not follow all the comments. I proceed to that work slowly. I recreated a new PR (CMSSW_10_0_X) in order to fix the conflicts!

**Comment by thomreis** [comment]
Please implement the comments here and close #21744 in that case.
The tests are also failing because you changed more than the necessary parts in DQM/L1TMonitor/python/L1TStage2_cff.py and reverted some changes made by others by mistake.

**Comment by jfernan2** [comment]
please test

File: DQM/L1TMonitor/plugins/SealModule.cc [hunk 1]
```diff
@@ -112,9 +115,3 @@ DEFINE_FWK_MODULE( L1GtHwValidation);
 
 #include "DQM/L1TMonitor/interface/L1ExtraDQM.h"
 DEFINE_FWK_MODULE(L1ExtraDQM);
-
```
Comment (thomreis) [inline DQM/L1TMonitor/plugins/SealModule.cc]: Why did you remove these lines?

**Comment by jfernan2** [comment]
please test

File: DQM/L1TMonitor/plugins/SealModule.cc [hunk 2]
```diff
@@ -33,6 +33,9 @@ DEFINE_FWK_MODULE(L1TStage2CaloLayer2);
 #include <DQM/L1TMonitor/interface/L1TStage2uGMT.h>
 DEFINE_FWK_MODULE(L1TStage2uGMT);
 
+#include <DQM/L1TMonitor/interface/L1TObjectsTiming.h>
```
Comment (thomreis) [inline DQM/L1TMonitor/plugins/SealModule.cc]: Remove this one since it is a duplicate from line 48.

File: DQM/L1TMonitor/plugins/SealModule.cc [hunk 3]
```diff
@@ -107,14 +113,8 @@ DEFINE_FWK_MODULE(L1TRate);
 #include "DQM/L1TMonitor/interface/L1TBPTX.h"
 DEFINE_FWK_MODULE(L1TBPTX);
 
-#include "DQM/L1TMonitor/interface/L1GtHwValidation.h"
-DEFINE_FWK_MODULE( L1GtHwValidation);
+//#include "DQM/L1TMonitor/interface/L1GtHwValidation.h"
```
Comment (thomreis) [inline DQM/L1TMonitor/plugins/SealModule.cc]: Do not comment this out.

File: DQM/L1TMonitor/plugins/SealModule.cc (hunk 1 above)
Comment (thomreis) [reply DQM/L1TMonitor/plugins/SealModule.cc]: These two still need to be added back. Compare to this file https://github.com/cms-sw/cmssw/blob/master/DQM/L1TMonitor/plugins/SealModule.cc

**Review by thomreis — COMMENTED** [review]
Still needs some changes in the sealModule.cc

**Comment by thomreis** [comment]
please abort test

This will fail again until the sealModule has not been fixed.

**Comment by thomreis** [comment]
please test

**Comment by thomreis** [comment]
please test

**Comment by thomreis** [comment]
This adds new DQM histograms. Therefore, the memory increase is expected.

**Comment by dmitrijus** [comment]
Hi, @thomreis 

How is this related to https://github.com/cms-sw/cmssw/pull/21844 ?
Which one should be merged first?

**Comment by thomreis** [comment]
Hi @dmitrijus 
better to merge this one first. I'll deal with eventual conflicts in #21844

**Comment by dmitrijus** [comment]
+1

**Comment by thomreis** [comment]
@davidlange6 can this be reviewed/merged please?

**Comment by fabiocos** [comment]
+1
